app/src/main/java/nu/yona/app/ui/message/MessageStickyRecyclerAdapter.java

```diff
@@ -64,63 +71,137 @@ public MessageItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType)
 	@Override
 	public void onBindViewHolder(MessageItemViewHolder holder, int position)
 	{
-		YonaMessage yonaObject = (YonaMessage) getItem(position);
+		currentYonaMessage = (YonaMessage) getItem(position);
+		currentMessageItemHolder = holder;
+		if (currentYonaMessage != null)
+		{
+			setUpMessageListItemTitle();
+			setUpMessageItemHolderDetails();
+			currentMessageItemHolder.deleteMsg.setTag(currentYonaMessage);
+			currentMessageItemHolder.messageContainer.setTag(currentYonaMessage);
+		}
+	}
 
-		if (yonaObject != null)
+	private void setUpMessageListItemTitle()
+	{
+		if (currentYonaMessage.getNotificationMessageEnum() != null && !TextUtils.isEmpty(currentYonaMessage.getNotificationMessageEnum().getUserMessage()))
```

Comment:
As far as I can see, ``currentYonaMessage.getNotificationMessageEnum().getUserMessage()`` never returns an empty string. Do you agree to that or am I overlooking something?

Reply:
That's not what's being compared here: The message returned in the JSON does not end up in the enum. As far as I can see, the user message in the enum is always a nonempty string.

Reply:
Sorry. I am being stupid here. I will correct it. Notification Enums are declared with user messages hard-coded.

Reply:
Corrected code. No needed all those checks as ```NotificationMessageEnum``` always has ```userMessage```.